packages/flutter/test/material/button_bar_test.dart

```diff
@@ -15,105 +15,279 @@ void main() {
     );
   });
 
-  testWidgets('ButtonBar has a min height of 52 when using ButtonBarLayoutBehavior.constrained', (WidgetTester tester) async {
-    await tester.pumpWidget(
-      SingleChildScrollView(
-        child: ListBody(
-          children: <Widget>[
-            ButtonTheme.bar(
-              layoutBehavior: ButtonBarLayoutBehavior.constrained,
-              child: const Directionality(
-                textDirection: TextDirection.ltr,
-                child: ButtonBar(
-                  children: <Widget>[
-                    SizedBox(width: 10.0, height: 10.0),
-                  ],
-                ),
-              ),
+  group('alignment', () {
+
+    testWidgets('default alignment is MainAxisAlignment.end', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        const MaterialApp(
+          home: ButtonBar(
+            children: <Widget>[
+              SizedBox(width: 10.0, height: 10.0),
+            ],
+          ),
+        )
+      );
+
+      final Finder child = find.byType(SizedBox);
+      expect(tester.getRect(child).left, 782.0);
```

Comment:
For tests like this it's helpful to include a comment that explains where the value comes from. In this case 782 = 800 - 10 - [something about the default padding].